infra/macos/m1-build/init-2.sh

```diff
@@ -0,0 +1,152 @@
+dscl . -create /Users/vsts NFSHomeDirectory /Users/vsts
+mkdir -p /Users/vsts
+chown vsts:staff /Users/vsts
+# END: macOS equivalent of useradd & groupadd
+
+echo "Done creating user vsts."
+
+# Homebrew must be installed by an admin, and I would rather not make vsts one.
+# So we need a little dance here.
+su -l admin <<'HOMEBREW_INSTALL'
+set -euo pipefail
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)" < /dev/null
+HOMEBREW_INSTALL
+# /usr/local itself belongs to root and cannot be changed, but we can change
+# the subdirs
+chown -R vsts /opt/homebrew
+chown -R vsts /opt/homebrew/*
+
+echo "Done installing Homebrew."
+
+# Install jq
+su -l vsts <<'TOOLS_INSTALL'
+set -euo pipefail
+cat << BASHRC >> /Users/vsts/.bashrc
+if [ -d /opt/homebrew/bin ]; then
+    export PATH="/opt/homebrew/bin:\$PATH"
+fi
+BASHRC
+/opt/homebrew/bin/brew install jq netcat xz
+TOOLS_INSTALL
+
+echo "Done installing tools through Homebrew."
+
+# Install Rosetta 
+/usr/sbin/softwareupdate --install-rosetta --agree-to-license
+
+# create /nix partition
+hdiutil create -size 60g -fs 'Case-sensitive APFS' -volname Nix -type SPARSE /System/Volumes/Data/Nix.dmg
+hdiutil attach /System/Volumes/Data/Nix.dmg.sparseimage -mountpoint /nix
+
+echo "Created /nix partition."
+
+## cache
+
+CACHE_SCRIPT=/Users/vsts/reset_caches.sh
+
+cat <<'RESET_CACHES' > $CACHE_SCRIPT
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+reset_cache() {
+    local file mount_point
+    file=$1
+    mount_point=$2
+
+    echo "Cleaning up '$mount_point'..."
+    if [ -d "$mount_point" ]; then
+        for pid in $(pgrep -a -f bazel | awk '{print $1}'); do
+            echo "Killing $pid..."
+            kill -s KILL $pid
+        done
+        for pid in $(lsof $mount_point | sed 1d | awk '{print $2}' | sort -u); do
+            echo "Killing $pid..."
+            kill -s KILL $pid
+        done
+        if hdiutil info | grep $mount_point; then
+            hdiutil detach "$mount_point"
+        fi
+        rm -rf $mount_point
+    fi
+
+    rm -f "${file}.sparseimage"
+    hdiutil create -size 200g -fs APFS -volname "$file" -type SPARSE "$file"
+    mkdir -p $mount_point
+    hdiutil attach "${file}.sparseimage" -mountpoint "$mount_point"
+    echo "Done."
+}
+
+reset_cache /var/tmp/bazel_cache.dmg /var/tmp/_bazel_vsts
+reset_cache /var/tmp/disk_cache.dmg /Users/vsts/.bazel-cache
+RESET_CACHES
+chown vsts:staff $CACHE_SCRIPT
+chmod +x $CACHE_SCRIPT
+
+su -l vsts <<END
+/Users/vsts/reset_caches.sh
+END
+
+echo "created cache partitions"
+
+# Note: installing Nix in single-user mode with /nix already existing and
+# writeable does not require sudoer access
+# Single-user install is no longer available in the 2.4+ installer, so we pin
+# _the installer_ to 2.3.16 then upgrade to nix 2.4.x.
+su -l vsts <<'END'
+set -euo pipefail
+export PATH="/usr/sbin:$PATH"
+bash <(curl -sSfL https://releases.nixos.org/nix/nix-2.3.16/install)
```

Comment:
```suggestion
(
cd $(mktemp -d)
curl https://releases.nixos.org/nix/nix-2.10.3/nix-2.10.3-aarch64-darwin.tar.xz > tarball
tar xzf tarball
cd nix-2.10.3-aarch64-darwin
printf '64d\nw\n' | ed -s install
./install --no-daemon
)
```

Looks like this somehow got lost on the machine 🤷